patches/@react-navigation+native+6.0.13.patch

```diff
@@ -0,0 +1,64 @@
+diff --git a/node_modules/@react-navigation/native/lib/module/index.js b/node_modules/@react-navigation/native/lib/module/index.js
+index c596c2b..737a54e 100644
+--- a/node_modules/@react-navigation/native/lib/module/index.js
++++ b/node_modules/@react-navigation/native/lib/module/index.js
+@@ -11,5 +11,6 @@ export { default as useLinkBuilder } from './useLinkBuilder';
+ export { default as useLinkProps } from './useLinkProps';
+ export { default as useLinkTo } from './useLinkTo';
+ export { default as useScrollToTop } from './useScrollToTop';
++export { default as stripKeysFromNavigationState } from './stripKeysFromNavigationState';
+ export * from '@react-navigation/core';
+ //# sourceMappingURL=index.js.map
+\ No newline at end of file
+diff --git a/node_modules/@react-navigation/native/lib/module/stripKeysFromNavigationState.js b/node_modules/@react-navigation/native/lib/module/stripKeysFromNavigationState.js
+new file mode 100644
+index 0000000..31ba628
+--- /dev/null
++++ b/node_modules/@react-navigation/native/lib/module/stripKeysFromNavigationState.js
+@@ -0,0 +1,17 @@
++// This function removes keys and some other properties from the navigation state 
++// in order to force react-navigation to regenerate those keys
++
++// keys - if we try to use the same keys it is going to break the navigation
```

Comment:
I feel like the natural question here is: Why does the navigation break? And what exactly breaks about it?

Reply:
Here is an example of what is happening if we don't remove keys. Unwanted jump from the settings screen to the about screen when resizing. It just works if we remove keys before.

https://user-images.githubusercontent.com/67908363/214559461-bd32365d-b0f7-40c2-aad1-c5ad0059c60b.mov

It looks like for some reason nested state is wiped out. I am not sure what exactly is happening under the hood but I could look for the exact reason if you want. But I am afraid that in the end we will be forced to get new keys anyway. 

It kind of makes sense to me to use different keys for different navigators. Those navigators give us a completely different structure of components and the keys are used as the keys for these components.

Reply:
>It looks like for some reason nested state is wiped out. I am not sure what exactly is happening under the hood but I could look for the exact reason if you want.

Yeah, I'd love to know the answer. It's great that we have some solution. But I'm having trouble understanding what the problem is exactly.

>It kind of makes sense to me to use different keys for different navigators. Those navigators give us a completely different structure of components and the keys are used as the keys for these components.

What makes sense about it?

**quick aside:** I am asking these questions because I anticipate that other stakeholders will ask them. The more robust an explanation we have the easier it is for us to get buy in on stuff or think of alternate solutions. Personally, I am really happy with the progress overall.